stock_picking_jobcan_wf_confirm/models/stock_picking_jobcan_wf_confirm.py

```diff
@@ -0,0 +1,102 @@
+import logging
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class StockPickingJobcanWfConfirm(models.Model):
+    _name = "stock.picking.jobcan.wf.confirm"
+    _description = "Stock Picking Jobcan Workflow Confirm"
+    _inherit = ["api.call.mixin"]
+
+    @api.model
+    def _scheduled_update(self):
+        _logger.info("Scheduled stock picking JobCan WF confirmation...")
+        pickings = self._get_assigned_pickings_with_jobcan_wf()
+        confirm_pickings = self._get_confirmed_pickings(pickings)
+
+        self._validate_confirmed_pickings(confirm_pickings)
+
+    def _get_assigned_pickings_with_jobcan_wf(self):
+        return self.env["stock.picking"].search(
+            [("state", "=", "assigned"), ("jobcan_wf_number", "!=", False)]
+        )
+
+    def _get_confirmed_pickings(self, pickings):
+        confirm_pickings = []
+        for picking in pickings:
+            try:
+                results = self._make_api_call_for_picking(picking)
+                if results and results[0].get("status") == "completed":
+                    confirm_pickings.append(picking)
+            except UserError as e:
+                self.notify_user(picking, str(e))
+        return confirm_pickings
+
+    def _make_api_call_for_picking(self, picking):
+        try:
+            params = {"id": picking.jobcan_wf_number}
+            response = picking.make_api_call(
+                "jobcan_wf", endpoint="v2/requests", params=params
+            )
+            response.raise_for_status()
+            return response.json().get("results", [])
+        except Exception as e:
+            _logger.error(
+                "API call failed for picking %s with error: %s", picking.id, str(e)
+            )
+            raise UserError(_("API call failed: {}".format(str(e)))) from e
```

Comment:
I think we don't need it because schedule action will run in background and this will not be displayed in UI.